src/integrationCapture.ts

```diff
@@ -0,0 +1,117 @@
+import { SDKEventCustomFlags } from './sdkRuntimeModels';
+import { Dictionary, queryStringParser, getCookies, getHref } from './utils';
+
+
+export const facebookClickIdProcessor = (clickId: string): string => {
+    if (!clickId) {
+        return '';
+    }
+
+    const clickIdParts = clickId.split('.');
+    if (clickIdParts.length === 4) {
+        return clickIdParts[3];
+    };
+
+    return clickId;
+};
+interface ProcessorFunction {
+    (clickId: string): string;
+}
+interface IntegrationIdMapping {
+    [key: string]: {
+        mappingValue: string;
+        processor?: ProcessorFunction;
+    };
+}
+
+const integrationMapping: IntegrationIdMapping = {
+    fbclid: {
+        mappingValue: 'Facebook.ClickId',
+        processor: facebookClickIdProcessor,
+    },
+    _fbp: {
+        mappingValue: 'Facebook.BrowserId',
+    },
+    _fbc: {
+        mappingValue: 'Facebook.ClickId',
+        processor: facebookClickIdProcessor,
+    },
+};
+
+export default class IntegrationCapture {
+    public clickIds: Dictionary<string>;
+
+    /**
+     * Captures Integration Ids from cookies and query params and stores them in clickIds object
+     */
+    public capture(): void {
+        const queryParams = this.captureQueryParams() || {};
+        const cookies = this.captureCookies() || {};
+
+        if (queryParams['fbclid'] && cookies['_fbc']) {
+            // Exclude fbclid if _fbc is present
```

Comment:
Seems like this should say `Exclude _fbc if fbclid is present`?

Reply:
Good catch. I think the naming of these two attributes confuses (or confounds) me.